File: src/RobocopyGUI.CSharp/Models/CopyOptionItem.cs
using System.ComponentModel;

namespace RobocopyGUI.Models
{
    /// <summary>
    /// コピーオプションのUI表示用アイテムクラス
    /// チェックボックスで選択可能なオプションを表現する
    /// </summary>
    public class CopyOptionItem : INotifyPropertyChanged
    {
        // オプションキー（robocopyのコマンドラインオプション）
        private string _optionKey;
        /// <summary>
        /// オプションキー（例: "/E", "/MIR"）
        /// </summary>
        public string OptionKey
        {
            get => _optionKey;
            set
            {
                _optionKey = value;
                OnPropertyChanged(nameof(OptionKey));
            }
        }

        // 表示名
        private string _displayName;
        /// <summary>
        /// UI上の表示名（日本語）
        /// </summary>
        public string DisplayName
        {
            get => _displayName;
            set
            {
                _displayName = value;
                OnPropertyChanged(nameof(DisplayName));
            }
        }

        // 説明
        private string _description;
        /// <summary>
        /// オプションの詳細説明（ツールチップ用）
        /// </summary>
        public string Description
        {
            get => _description;
            set
            {
                _description = value;
                OnPropertyChanged(nameof(Description));
            }
        }

        // 選択状態
        private bool _isSelected;
        /// <summary>
        /// 選択されているかどうか
        /// </summary>
        public bool IsSelected
        {
            get => _isSelected;
            set
            {
                if (_isSelected != value)
                {
                    _isSelected = value;
                    OnPropertyChanged(nameof(IsSelected));
                }
            }
        }

        /// <summary>
        /// プロパティ変更通知イベント
        /// </summary>
        public event PropertyChangedEventHandler PropertyChanged;

        /// <summary>
        /// プロパティ変更を通知
        /// </summary>
        /// <param name="propertyName">変更されたプロパティ名</param>
        protected virtual void OnPropertyChanged(string propertyName)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}

File: src/RobocopyGUI.VB/Models/CopyOptionItem.vb
Imports System.ComponentModel

Namespace Models

    ''' <summary>
    ''' コピーオプションのUI表示用アイテムクラス
    ''' チェックボックスで選択可能なオプションを表現する
    ''' </summary>
    Public Class CopyOptionItem
        Implements INotifyPropertyChanged

        ' オプションキー（robocopyのコマンドラインオプション）
        Private _optionKey As String
        ''' <summary>
        ''' オプションキー（例: "/E", "/MIR"）
        ''' </summary>
        Public Property OptionKey As String
            Get
                Return _optionKey
            End Get
            Set(value As String)
                _optionKey = value
                OnPropertyChanged(NameOf(OptionKey))
            End Set
        End Property

        ' 表示名
        Private _displayName As String
        ''' <summary>
        ''' UI上の表示名（日本語）
        ''' </summary>
        Public Property DisplayName As String
            Get
                Return _displayName
            End Get
            Set(value As String)
                _displayName = value
                OnPropertyChanged(NameOf(DisplayName))
            End Set
        End Property

        ' 説明
        Private _description As String
        ''' <summary>
        ''' オプションの詳細説明（ツールチップ用）
        ''' </summary>
        Public Property Description As String
            Get
                Return _description
            End Get
            Set(value As String)
                _description = value
                OnPropertyChanged(NameOf(Description))
            End Set
        End Property

        ' 選択状態
        Private _isSelected As Boolean
        ''' <summary>
        ''' 選択されているかどうか
        ''' </summary>
        Public Property IsSelected As Boolean
            Get
                Return _isSelected
            End Get
            Set(value As Boolean)
                If _isSelected <> value Then
                    _isSelected = value
                    OnPropertyChanged(NameOf(IsSelected))
                End If
            End Set
        End Property

        ''' <summary>
        ''' プロパティ変更通知イベント
        ''' </summary>
        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        ''' <summary>
        ''' プロパティ変更を通知
        ''' </summary>
        ''' <param name="propertyName">変更されたプロパティ名</param>
        Protected Overridable Sub OnPropertyChanged(propertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub

    End Class

End Namespace
